internal/app/connectconformance/test_case_library_test.go

```diff
@@ -0,0 +1,292 @@
+		cases  map[serverInstance][]string
+	}{
+		{
+			name: "client mode",
+			config: []configCase{
+				{
+					Version:     conformancev1alpha1.HTTPVersion_HTTP_VERSION_1,
+					Protocol:    conformancev1alpha1.Protocol_PROTOCOL_CONNECT,
+					Codec:       conformancev1alpha1.Codec_CODEC_PROTO,
+					Compression: conformancev1alpha1.Compression_COMPRESSION_IDENTITY,
+					StreamType:  conformancev1alpha1.StreamType_STREAM_TYPE_UNARY,
+				},
+				{
+					Version:     conformancev1alpha1.HTTPVersion_HTTP_VERSION_1,
+					Protocol:    conformancev1alpha1.Protocol_PROTOCOL_CONNECT,
+					Codec:       conformancev1alpha1.Codec_CODEC_PROTO,
+					Compression: conformancev1alpha1.Compression_COMPRESSION_IDENTITY,
+					StreamType:  conformancev1alpha1.StreamType_STREAM_TYPE_UNARY,
+					UseTLS:      true,
+				},
+				{
+					Version:       conformancev1alpha1.HTTPVersion_HTTP_VERSION_1,
+					Protocol:      conformancev1alpha1.Protocol_PROTOCOL_CONNECT,
+					Codec:         conformancev1alpha1.Codec_CODEC_PROTO,
+					Compression:   conformancev1alpha1.Compression_COMPRESSION_IDENTITY,
+					StreamType:    conformancev1alpha1.StreamType_STREAM_TYPE_UNARY,
+					UseConnectGET: true,
+				},
+				{
+					Version:            conformancev1alpha1.HTTPVersion_HTTP_VERSION_1,
+					Protocol:           conformancev1alpha1.Protocol_PROTOCOL_CONNECT,
+					Codec:              conformancev1alpha1.Codec_CODEC_PROTO,
+					Compression:        conformancev1alpha1.Compression_COMPRESSION_IDENTITY,
+					StreamType:         conformancev1alpha1.StreamType_STREAM_TYPE_UNARY,
+					ConnectVersionMode: conformancev1alpha1.TestSuite_CONNECT_VERSION_MODE_REQUIRE,
+				},
+				{
+					Version:     conformancev1alpha1.HTTPVersion_HTTP_VERSION_2,
+					Protocol:    conformancev1alpha1.Protocol_PROTOCOL_GRPC,
+					Codec:       conformancev1alpha1.Codec_CODEC_PROTO,
+					Compression: conformancev1alpha1.Compression_COMPRESSION_IDENTITY,
+					StreamType:  conformancev1alpha1.StreamType_STREAM_TYPE_FULL_DUPLEX_BIDI_STREAM,
+				},
+			},
+			mode: conformancev1alpha1.TestSuite_TEST_MODE_CLIENT,
+			cases: map[serverInstance][]string{
+				{
+					protocol:    conformancev1alpha1.Protocol_PROTOCOL_CONNECT,
+					httpVersion: conformancev1alpha1.HTTPVersion_HTTP_VERSION_1,
+					useTLS:      false,
+				}: {
+					"Basic/HTTPVersion:1/Protocol:PROTOCOL_CONNECT/Codec:CODEC_PROTO/Compression:COMPRESSION_IDENTITY/basic-unary",
+					"Connect GET/HTTPVersion:1/Codec:CODEC_PROTO/Compression:COMPRESSION_IDENTITY/connect-get-unary",
+					"Connect Version Required (client)/HTTPVersion:1/Codec:CODEC_PROTO/Compression:COMPRESSION_IDENTITY/unary-without-connect-version-header",
+				},
+				{
+					protocol:    conformancev1alpha1.Protocol_PROTOCOL_CONNECT,
+					httpVersion: conformancev1alpha1.HTTPVersion_HTTP_VERSION_1,
+					useTLS:      true,
+				}: {
+					"TLS/HTTPVersion:1/Protocol:PROTOCOL_CONNECT/Codec:CODEC_PROTO/Compression:COMPRESSION_IDENTITY/tls-unary",
```

Comment:
Open question here: in this test, we can see that it will only run the TLS-**specific** test case against the TLS server. My thinking is that we'd define just basic "success" RPCs over TLS as sufficient evidence that TLS is functioning.

But another interpretation might be to run _all_ test cases through both plaintext and TLS, when TLS is supported. Overkill? WDYT?

Reply:
My opinion would be to leave it out for now. The ability to serve plaintext over a secure server seems like more of an outlier/specialty case so it feels like overkill. We could always add them in later if we think it's worthwhile to test. Just my 2 cents.